src/main/java/io/javalin/plugin/openapi/annotations/AnnotationApi.kt

```diff
@@ -0,0 +1,75 @@
+package io.javalin.plugin.openapi.annotations
+
+import kotlin.reflect.KClass
+
+/**
+ * Provide metadata for the generation of the open api documentation to the annotated Handler.
+ */
+@Target(AnnotationTarget.FUNCTION)
+annotation class OpenApi(
+        val summary: String = NULL_STRING,
+        val description: String = NULL_STRING,
+        val operationId: String = NULL_STRING,
+        val deprecated: Boolean = false,
+        val tags: Array<String> = [],
+        val cookies: Array<OpenApiParam> = [],
+        val headers: Array<OpenApiParam> = [],
+        val pathParams: Array<OpenApiParam> = [],
+        val queryParams: Array<OpenApiParam> = [],
+        val requestBodies: Array<OpenApiRequestBody> = [],
+        val fileUploads: Array<OpenApiFileUpload> = [],
+        val responses: Array<OpenApiResponse> = []
+)
+
+@Target()
+annotation class OpenApiResponse(
+        val status: String,
+
+        val returnType: KClass<*> = NULL_CLASS::class,
+
+        val contentType: String = ContentType.AUTODETECT,
+
+        /** Whenever the returnType should be wrapped in an array */
+        val isArray: Boolean = false,
+
+        val description: String = NULL_STRING
+)
+
+@Target()
+annotation class OpenApiParam(
+        val name: String,
+        val type: KClass<*> = String::class,
+        val description: String = NULL_STRING,
+        val deprecated: Boolean = false,
+        val required: Boolean = false,
+        val allowEmptyValue: Boolean = false
+)
+
+@Target()
+annotation class OpenApiRequestBody(
+        val type: KClass<*>,
+        val contentType: String = ContentType.AUTODETECT,
+        val required: Boolean = false,
+        val description: String = NULL_STRING
+)
+
+@Target()
+annotation class OpenApiFileUpload(
+        val name: String,
+        val isArray: Boolean = false,
+        val description: String = NULL_STRING,
+        val required: Boolean = false
+)
+
+
+/** Null string because annotations do not support null values */
+const val NULL_STRING = "io.javalin.plugin.openapi This string represents a null value and shouldn't be used"
```

Comment:
`io.javalin.plugin.openapi This string represents a null value and shouldn't be used`. What's going on here? Why is the package part of the string?

Reply:
I used it as a prefix to avoid duplications. But I will remove it to avoid confusion.